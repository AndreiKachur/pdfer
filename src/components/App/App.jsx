import { useState } from 'react';

import Main from '../Main/Main';
import Screensaver from '../Screensaver/Screensaver';

const App = () => {
  const [isScreensaver, setIsScreensaver] = useState(true);

  if (isScreensaver) {
    return <Screensaver setIsScreensaver={setIsScreensaver} />;
  }

  return <Main />;
};

export default App;
