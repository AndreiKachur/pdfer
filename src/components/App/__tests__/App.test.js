import { render } from '@testing-library/react';

import App from '../App';

describe('App component', () => {
  it('should render a valid snapshot', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
