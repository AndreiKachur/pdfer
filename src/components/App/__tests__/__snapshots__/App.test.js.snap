// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render a valid snapshot 1`] = `
<DocumentFragment>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header header"
    >
      <div
        class="logo"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.99787498,8.99999999 L4.99787498,0.999999992 L19.4999998,0.999999992 L22.9999998,4.50000005 L23,23 L4,23 M18,1 L18,6 L23,6 M3,12 L3.24999995,12 L4.49999995,12 C6.5,12 6.75,13.25 6.75,14 C6.75,14.75 6.5,16 4.49999995,16 L3.24999995,16 L3.24999995,18 L3,17.9999999 L3,12 Z M9.5,18 L9.5,12 C9.5,12 10.4473684,12 11.2052633,12 C12.3421053,12 13.5,12.5 13.5,15 C13.5,17.5 12.3421053,18 11.2052633,18 C10.4473684,18 9.5,18 9.5,18 Z M16.5,19 L16.5,12 L20.5,12 M16.5,15.5 L19.5,15.5"
            fill="none"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
        <p
          class="title"
        >
          Pdfer
        </p>
      </div>
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark menu"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-overflow-item ant-menu-item"
          data-menu-id="rc-menu-uuid-test-themeMode"
          role="menuitem"
          style="opacity: 1; order: 0;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <button
              aria-checked="false"
              class="ant-switch"
              role="switch"
              type="button"
            >
              <div
                class="ant-switch-handle"
              />
              <span
                class="ant-switch-inner"
              />
            </button>
            <p
              class="title"
            >
              Dark
            </p>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item"
          role="menuitem"
          style="opacity: 1; order: 1;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M13.353 1.146l1.5 1.5L15 3v11.5l-.5.5h-13l-.5-.5v-13l.5-.5H13l.353.146zM2 2v12h12V3.208L12.793 2H11v4H4V2H2zm6 0v3h2V2H8z"
                fill-rule="evenodd"
              />
            </svg>
            <p
              class="title"
            >
              Save
            </p>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item"
          role="menuitem"
          style="opacity: 1; order: 2;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1,17 L1,23 L23,23 L23,17 M12,2 L12,19 M5,12 L12,19 L19,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
            <p
              class="title"
            >
              Load
            </p>
          </span>
        </li>
        <li
          aria-hidden="true"
          class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
        >
          <div
            aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-rc-menu-more"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </header>
    <main
      class="ant-layout-content content"
    >
      <div
        class="siteLayoutContent"
      >
        Content
      </div>
    </main>
    <footer
      class="ant-layout-footer footer"
    >
      Â©2022 Created by Andrey Kachur
    </footer>
  </section>
</DocumentFragment>
`;
